tests/test_de.py

```diff
@@ -31,17 +31,32 @@ def get_annotation_labels(metadata, annotation, de_cells):
     return unique_labels.tolist()
 
 
-def find_expected_files(labels, cluster_name, annotation, scope, method):
+def find_expected_files(labels, cluster_name, test_config):
     """Check that files were created for all expected annotation labels"""
     found = []
     sanitized_cluster_name = DifferentialExpression.sanitize_string(cluster_name)
-    sanitized_annotation = DifferentialExpression.sanitize_string(annotation)
-    for label in labels:
-        sanitized_label = DifferentialExpression.sanitize_string(label)
-        expected_file = f"{sanitized_cluster_name}--{sanitized_annotation}--{sanitized_label}--{scope}--{method}.tsv"
+    sanitized_annotation = DifferentialExpression.sanitize_string(test_config.get("test_annotation"))
+    de_type =  test_config.get("de_type")
+    method = test_config.get("test_method")
+    annot_scope = test_config.get("test_scope")
+    if de_type == "rest":
+        for label in labels:
+            sanitized_label = DifferentialExpression.sanitize_string(label)
+            expected_file = f"{sanitized_cluster_name}--{sanitized_annotation}--{sanitized_label}--{annot_scope}--{method}.tsv"
+            print(f'expected_file, {expected_file}')
+            assert os.path.exists(expected_file)
+            found.append(expected_file)
+        return found
+    elif de_type == "pairwise":
+        # rank_genes_groups accepts a list. For SCP pairwise, should be a list with one item
+        # converting list to string for incorporation into result filename
+        group1 = DifferentialExpression.sanitize_string(''.join(test_config["group1"]))
+        group2 = DifferentialExpression.sanitize_string(test_config["group2"])
+        expected_file = f'{sanitized_cluster_name}--{sanitized_annotation}--{group1}--{group2}--{annot_scope}--{method}.tsv'
+        print(f'expected_file, {expected_file}')
         assert os.path.exists(expected_file)
         found.append(expected_file)
-    return found
+        return found
```

Comment:
Having `return found` after ending the `elif` block, and omitting `return found` in the `if` block, seems more conventional.

Reply:
`e6287de`